embedded-value/src/main/java/com/iluwatar/embedded/value/DataSource.java

```diff
@@ -0,0 +1,217 @@
+  
+  /**
+   * {@summary} 
+   * Establish connection to database.
+   */
+  public DataSource() {
+    try {
+      conn = DriverManager.getConnection(JDBC_URL);
+      LOGGER.info("Connected to H2 in-memory database: " + conn.getCatalog());
+    } catch (SQLException e) {
+      LOGGER.error(e.getLocalizedMessage(), e.getCause());
+    }
+  }
+
+  @Override
+  public boolean createSchema() {
+    try {
+      createschema = conn.createStatement();
+      createschema.execute(CREATE_SCHEMA);
+      insertIntoOrders = conn.prepareStatement(INSERT_ORDER, PreparedStatement.RETURN_GENERATED_KEYS);
+      getschema = conn.createStatement();
+      queryOrders = conn.createStatement();
+      removeorder = conn.prepareStatement(REMOVE_ORDER);
+      queyOrderByID = conn.prepareStatement(QUERY_ORDER);
+      deleteschema = conn.createStatement();
+      createschema.close();
+    } catch (SQLException e) {
+      LOGGER.error(e.getLocalizedMessage(), e.getCause());
+      return false;
+    }
+    return true;
+  }
+
+  @Override
+  public String getSchema() {
+    try {
+      var resultSet = getschema.executeQuery(GET_SCHEMA);
+      var sb = new StringBuilder();
+      while (resultSet.next()) {
+        sb.append("Col name: " + resultSet.getString(1) + ",  Col type: " + resultSet.getString(2) + "\n");
+      }
+      getschema.close();
+      return sb.toString();
+    } catch (Exception e) {
+      LOGGER.error("Error in retrieving schema: {}", e.getLocalizedMessage(), e.getCause());
+    }
+    return "Schema unavailable";
+  }
+  
+  @Override
+  public boolean insertOrder(Order order) {
+    try {
+      conn.setAutoCommit(false);
+      insertIntoOrders.setString(1, order.getItem());
+      insertIntoOrders.setString(2, order.getOrderedBy());
+      var address = order.getShippingAddress();
+      insertIntoOrders.setString(3, address.getCity());
+      insertIntoOrders.setString(4, address.getState());
+      insertIntoOrders.setString(5, address.getPincode());
+      var affectedRows = insertIntoOrders.executeUpdate();
+      if (affectedRows == 1) {
+        var rs = insertIntoOrders.getGeneratedKeys();
+        rs.last();
+        order.setId(rs.getInt(1));
+        conn.commit();
+        LOGGER.info("Inserted: {}", order);
+      } else {
+        conn.rollback();
+      }
+    } catch (Exception e) {
+      LOGGER.error(e.getLocalizedMessage());
+    } finally {
+      try {
+        conn.setAutoCommit(true);
+      } catch (SQLException e) {
+        LOGGER.error(e.getLocalizedMessage());
+      }
+    }
+    return true;
+  }
+
+  @Override
+  public Stream<Order> queryOrders() {
+    var ordersList = new ArrayList<Order>();
+    try (var rSet = queryOrders.executeQuery(QUERY_ORDERS)) {
+      while (rSet.next()) {
+        var order = new Order(rSet.getString(2), rSet.getString(3),
+            new ShippingAddress(rSet.getString(4), rSet.getString(5),
+            rSet.getString(6)));
+        order.setId(rSet.getInt(1));
```

Comment:
Why not using the constructor of Order to supplier the id instead of using a Setter?